components/net/fetch/methods.rs

```diff
@@ -218,16 +220,16 @@ pub async fn main_fetch(
     let mut response = None;
 
```

Comment:
Can you add spec step 1 here?
```
// Step 1. Let request be fetchParams’s request.
let request = &mut fetch_params.request;
```
and replace the usages of `fetch_params.request` everywhere in the function?

The purpose of these changes is to bring the fetch code closer to the spec, so let's make sure to actually follow it (:

Reply:
haha yea I got a bit ahead of myself and assumed the `fetch` function also took fetchParams.

Reply:
I ended up changing all usages of `request` to `fetch_params.request` because of mutable borrowing issues. In the `main_fetch` function there are calls to `scheme_fetch` and `http_fetch` that take `&mut fetch_params`. If I did
```
let request = &mut fetch_params.request;
```
then I'm not able to pass in a `&mut fetch_params` to those `fetch` functions without double borrowing. Is there a good way I can work around that?

Reply:
If taking a reference to the fetchparams.request field causes borrowing issues then it's fine to keep the code as - is. Just make sure to note that down somewhere in the source code, so people don't get confused.